core/src/dird/ua_select.cc

```diff
@@ -862,23 +865,35 @@ bool SelectStorageDbr(UaContext* ua, StorageDbRecord* sr, const char* argk)
 }
 
 // Select a Media (Volume) record from the database
-bool SelectMediaDbr(UaContext* ua, MediaDbRecord* mr)
+bool SelectMediaDbr(UaContext* ua, MediaDbRecord* mr, const char* volumename)
```

Comment:
I think this function already did way too much before you added another feature.
The problem is `SelectPoolAndMediaDbr()` which clearly violates SRP (the name already gives it away).
I'd refactor `SelectPoolAndMediaDbr()` into `SelectPoolForMedia()` by just removing the call to `SelectMediaDbr()` at the beginning and requiring the caller to have `MediaDbRecord*` already filled.
You'll then need to add a call to call `SelectMediaDbr(ua, mr)` before the one existing use of `SelectPoolAndMediaDbr()`.

After refactoring like that, you can then focus on what you really need and implement a new function that will do the following:
* reset `mr`
* fill `mr->VolumeName`
* call `ua->db->GetMediaRecord(ua->jcr, mr)`

I guess something like this should be sufficient (and way easier to understand):
```
bool SelectMediaByName(UaContext* ua, MediaDbRecord* mr, const char* volumename) {
  new (mr) MediaDbRecord();
  bstrncpy(mr->VolumeName, volumename, sizeof(mr->VolumeName));
  if (!ua->db->GetMediaRecord(ua->jcr, mr)) {
    ua->ErrorMsg("%s", ua->db->strerror());
    return false;
  }
  return true;
}
```
If you want to go the extra mile, you can add a new constructor `MediaDbRecord(const char*)` so that you can pass `volumename` as part of the placement-new.